exploration/number-selection.md

```diff
@@ -92,16 +95,78 @@ ICU MF1 messages using `plural` and `selectordinal` should be representable in M
+or is some serialization of a number that the implementation recognizes.
+
+> For example, in Java, any subclass of `java.lang.Number` plus the primitive
+> types (`boolean`, `byte`, `short`, `int`, `long`, `float`, `double`, etc.) 
+> might be considered as the "implementation-defined numeric types".
+> Implementations in other programming languages would define different types
+> or classes according to their local needs.
+
+Implementations SHOULD treat numeric literals that produce parse errors as
+non-numeric string literals.
+That is, numeric selection MUST emit a _Selection Error_ for these values.
+
+The function `:number` has the following options:
+- `select` (`plural`, `ordinal`, `exact`; default: `plural`)
+- `compactDisplay` (`short`, `long`; default: `short`)
+- `currency` (ISO 4712 currency code)
+- `currencyDisplay` (`symbol` `narrowSymbol` `code` `name`; default: `symbol`)
+- `currencySign` (`accounting`, `standard`; default: `standard`)
+- `notation` (`standard` `scientific` `engineering` `compact`; default: `standard`)
+- `numberingSystem` (arab arabext bali beng deva fullwide gujr guru hanidec khmr knda laoo latn 
+   limb mlym mong mymr orya tamldec telu thai tibt)
+- `signDisplay` (`auto` `always` `exceptZero` `never`; default=`auto`)
+- `style` (`decimal` `currency` `percent` `unit`; default=`decimal`)
+- `unit` (anything not empty)
+- `unitDisplay` (`long` `short` `narrow`; default=`short`)
+- `minimumIntegerDigits`, (positive integer, default: `1`)
+- `minimumFractionDigits`, (positive integer)
+- `maximumFractionDigits`, (positive integer)
+- `minimumSignificantDigits`, (positive integer, default: `1`)
+- `maximumSignificantDigits`, (positive integer, default: `21`)
+
+> [!NOTE]
+> We have a requirement for `offset` but have not provided for it yet??
+
+### Selection
+
+When performing selection, the resolved value of the `operand` is compared to 
+each of the variant keys.
+The result of such a comparison can be one of the following:
+
+- If the value of the key does not match the production `number-literal`
+  or is not one of the plural/ordinal keywords, the key is invalid;
+  return a _Selection Error_.
+- If the value of the key matches the production `number-literal`:
+  - If the parsed value of the key is an [exact match](#determining-exact-literal-match)
+    of the value of the operand, then the key matches the selector with the highest
+    level match supported by the selector.
+- Else, if the value of the key is a keyword and the value of `select` is not `exact`:
+  - If the keyword value of running plural(operand) or ordinal(operand)
+    equals the key, the key matches the selector with a level of match
+    lower than the highest level, but higher than the default.
+> [!NOTE]
+> The default keyword `other` can produce this level of match.
+- Else, if the value of the key is `*`
+  - The key matches the selector at the lowest (default) level of match.
```

Comment:
This never happens, because a `*` is never compared to the selector value. Its preferential position is automatic.
```suggestion
```

Reply:
I know that you've written the algorithm to never do the comparison. I think that might be wrong. I think a selector should be exposed to all of the keys and decide how to sort `*` into the mix. Among other things, in `:number` this allows the plural rule `other` to be a true synonym to `*`

Reply:
This is why I called out during our last call reviewing the current selection algorithm, because if there is any desire to change it, we really should have a PR available for review ASAP. I don't think such a change would be necessary or beneficial, but I'd be open and interested to review specific changes.

My own sense is that the current algorithm is capable of supporting any actual selection behaviour, and that further changes would add unnecessary complexity.

Reply:
My instinct was to propose changes to that algorithm first, which is why this took so long to start writing. In the end I decided to work on this as an aid to proposing changes there. I see what you're saying. I'm re-writing the algorithm "your" way this morning and we'll see if that makes more sense.